src/profile-logic/sanitize.js

```diff
@@ -250,6 +251,30 @@ function sanitizeThreadPII(
       ) {
         // Sanitize all the name fields of text markers in case they contain URLs.
         markerTable.data[i] = sanitizeTextMarker(currentMarker);
+        // Re-assign the value of currentMarker as the marker may be
+        // sanitized again to remove extension ids.
+        currentMarker = markerTable.data[i];
+      }
+
+      if (
+        PIIToBeRemoved.shouldRemoveExtensions &&
+        currentMarker &&
+        currentMarker.type &&
+        currentMarker.type === 'Text'
+      ) {
+        const markerName = stringArray[markerTable.name[i]];
+        // Sanitize extension ids out of known extension markers.
+        if (markerName === 'ExtensionParent') {
+          markerTable.data[i] = {
+            ...currentMarker,
+            name: currentMarker.name.replace(/^.*, (api_(call|event): )/, '$1'),
+          };
+        } else if (markerName === 'Extension Suspend') {
```

Comment:
Oh, looks like `Extension Suspend` was an existing marker and we weren't sanitizing it properly. Thanks for adding this too.